worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java

```diff
@@ -330,6 +330,11 @@ public ServingState currentServingState() {
   public void switchServingState() {
     ServingState lastState = servingState;
     servingState = currentServingState();
+    if (servingState == lastState && servingState == ServingState.NONE_PAUSED) {
+      // this means that there is no memory pressure here
+      // skip the rest check logic of memory checker
+      return;
+    }
     logger.info("Serving state transformed from {} to {}", lastState, servingState);
```

Comment:
how about simply wrapping the logging stmt with `if (servingState != lastState)`?

this is not the perf-sensitive code path, better to avoid introducing additional control flow

Reply:
Sounds good to me. I'll update this PR.